hydroflow_lang/src/graph/ops/join_fused.rs

```diff
@@ -0,0 +1,333 @@
+use proc_macro2::{Ident, TokenStream};
+use quote::{quote_spanned, ToTokens};
+use syn::spanned::Spanned;
+use syn::{parse_quote, Expr, ExprCall};
+
+use super::{
+    DelayType, FlowProperties, FlowPropertyVal, OperatorCategory, OperatorConstraints,
+    OperatorWriteOutput, Persistence, WriteContextArgs, RANGE_0, RANGE_1,
+};
+use crate::diagnostic::{Diagnostic, Level};
+use crate::graph::{OpInstGenerics, OperatorInstance};
+
+/// > 2 input streams of type <(K, V1)> and <(K, V2)>, 1 output stream of type <(K, (V1, V2))>
+///
+/// `join_fused` takes two arguments, they are the left hand side and right hand side reducing/folding operation respectively.
+/// The syntax is a bit unique for them, but it is similar to specifying an enum in rust. For example:
+/// ```hydroflow,ignore
+/// join_fused(Fold(|| "default value", |x, y| FoldingFunction(x, y)), Reduce(|x, y| ReducingFunction(x, y)))
```

Comment:
I was confused by this on first read. 

You could clarify what you mean by "similar to specifying an enum" to say "for each input to the join you specify one of the following three options: ... `Reduce` (if the input type is the same as the accum type), `Fold` (if the input type is different than the accum type but has a sensible default) or `FoldFrom` (if the input type is different than the accum and needs to be derived from the first data item passed in)

Then before the example say: "Consider a case where the LHS of the join requires a `Fold`, and the RHS requires a `Reduce`".